src/checkLinuxWebAppDownDetector.ts

```diff
@@ -0,0 +1,175 @@
+            detectorErrorMessage = await getLinuxDetectorError(requestOptions, deployResultTime, node.root.client.fullName);
+            // poll every minute
+            await delay(1000 * 60);
+        } while (!detectorErrorMessage);
+
+        await ext.ui.showWarningMessage(detectorErrorMessage, { title: 'Open in Portal' });
+        const portalDeeplink = `${node.root.environment.portalUrl}/?websitesextension_ext=asd.featurePath%3Ddetectors%2F${detectorId}#@microsoft.onmicrosoft.com/resource/${node.root.client.id}/troubleshoot`;
+        await openUrl(portalDeeplink);
+        context.telemetry.properties.didClick = 'true';
+    });
+}
+
+async function signRequest(req: WebResource, cred: ServiceClientCredentials): Promise<void> {
+    await new Promise((resolve: () => void, reject: (err: Error) => void): void => {
+        cred.signRequest(req, (err: Error | undefined) => {
+            if (err) {
+                reject(err);
+            } else {
+                resolve();
+            }
+        });
+    });
+}
+
+async function getLinuxDetectorError(requestOptions: WebResource, deployResultTime: Date, siteName: string): Promise<string | undefined> {
+    const requestPromise = <(options: RequestOptions | string | URL) => Promise<IncomingMessage>><Function>requestP;
+    const detectorResponse: string = <string><unknown>(await requestPromise(requestOptions));
+    const responseJson: detectorResponseJSON = <detectorResponseJSON>JSON.parse(detectorResponse);
+
+    const datasets: detectorDataset[] = responseJson.properties.dataset;
+    const vsCodeIntegration: string = 'Latest time seen by detector. To be used in VSCode integration.';
+    const timestampTable: detectorTable | undefined = findTableByRowValue(datasets, vsCodeIntegration);
+    if (!timestampTable) {
+        // if we can't find the timestamp, exit and try again
+        return undefined;
+    }
+
+    const timestamp = new Date(getValuesByColumnName(timestampTable, ColumnName.value)[0]);
+    if (validateTimestamp(timestamp, deployResultTime)) {
+
+        const criticalErrorTable: detectorTable | undefined = findTableByRowValue(datasets, 'Critical');
+        const failureCountTable: detectorTable | undefined = findTableByColumnName(datasets, ColumnName.failureCount);
+        if (!criticalErrorTable || !failureCountTable) {
+            return undefined;
+        }
+
+        const errorMessages: string[] = getValuesByColumnName(criticalErrorTable, ColumnName.value);
+        const failureCount: string = getValuesByColumnName(failureCountTable, ColumnName.failureCount)[0];
+
+        return `Critical insights found for "${siteName}": ${failureCount} as of ${new Date(timestamp).toLocaleString()}. Critical insight preview: "${errorMessages[0]}". Click here to access "App Service Diagnostics" for recommendations.`;
+    }
+
+    return undefined;
+}
+
+export function validateTimestamp(detectorTime: Date, deployResultTime: Date): boolean {
```

Comment:
So this checks that the detector and deploy result time are within a minute of each other? Can you explain why it does that? Seems like it should pick the first detector that exists after the deploy, regardless of how close in time they are